### PR TITLE
Cats effect module

### PR DESCRIPTION
**Pull request by Slakah**
Implements #200 .

Ammonite example of `IOCaseApp`:
```scala
@ import cats.effect._
@ import caseapp.cats._
@ import caseapp._
@ case class Options(foo: String)
@
object Main extends IOCaseApp[Options] {
  override def run(options: Options, remainingArgs: RemainingArgs) = IO {
    println(options)
    ExitCode.Success
  }
}
@ Main.main(Array("--help"))

Usage: [options]
  --usage  <bool>
        Print usage and exit
  --help | -h  <bool>
        Print help message and exit
  --foo  <string>
@ Main.main(Array("--usage"))
Usage: [options]
@ Main.main(Array("--foo", "foo"))
Options(foo)
```

Ammonite example of `IOCommandApp`:

```scala
@ import cats.effect._
@ import caseapp.cats._
@ import caseapp._
@ object Command {
    sealed trait Command
    case class Foo(foo: String) extends Command
    case class Bar(bar: Int) extends Command
  }
@ import Command._
@
object Main extends IOCommandApp[Command] {
  override def run(command: Command, remainingArgs: RemainingArgs) = IO {
    println(command)
    ExitCode.Success
  }
}
@ Main.main(Array("--help"))
None.type
Usage: none.type [options] [command] [command-options]

Available commands: bar, foo

Type  none.type command --help  for help on an individual command
@ Main.main(Array("--usage"))
Usage: none.type [options] [command] [command-options]
Available commands: bar, foo

Type  none.type command --usage  for usage of an individual command
@ Main.main(Array("foo", "--help"))
Command: foo
Usage: none.type foo
  --foo  <string>
@ Main.main(Array("foo", "--foo", "foo"))
Foo(foo)
@ Main.main(Array("bar", "--bar", "1"))
Bar(1)
@ Main.main(Array("bar", "--bar", "foo"))
Argument --bar: Malformed integer: foo
```

## TODO

- [x] Implementation
- [x] Tests - @alexarchambault some guidance here would be greatly appreciated, as I couldn't find any tests for `CaseApp`/`CommandApp`/etc. to copy from. Maybe a quick test which runs the jar, and looks at stdout, stderr and the exit code would be nice?
- [x] Documentation - add something quick to the readme

### CODE REVIEW

**Comment by Slakah** [comment]
> They can create some object … extends IOCaseApp[…], then call the run method of those, run the IO, via a Future say (utest accepts a Future as last value), and check the value they return.

Cool sounds good, the bit I was struggling with was how to capture stdout/stderr. I'd be tempted to allow stdout + stderr to be overridden, and to store the output in a mutable buffer.

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings" % "3.0.0")
 addSbtPlugin("com.geirsson" % "sbt-ci-release" % "1.5.3")
 addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.7.0")
-addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.0.1")
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.1.0")
```
Comment (Slakah) [inline project/plugins.sbt:4]: Updated there were errors with the https://github.com/alexarchambault/case-app/pull/202/files#diff-2509e699b354635ff26b61546c07e5d5R10 when running js tests.

Can either revert this version bump and make the tests JVM specific, or keep as is.

**Comment by Slakah** [comment]
@alexarchambault , happy with the PR now, let me know if you want me to make any other changes.

File: tests/shared/src/test/scala/caseapp/CaseAppTests.scala [hunk 2]
```diff
@@ -8,12 +8,12 @@ import shapeless.{Inl, Inr}
 import utest._
 import caseapp.core.util.Formatter
 
-object Tests extends TestSuite {
+object CaseAppTests extends TestSuite {
```
Comment (Slakah) [inline tests/shared/src/test/scala/caseapp/CaseAppTests.scala:11]: Renamed as this conflicted with https://github.com/lihaoyi/utest/blob/master/utest/src/utest/Tests.scala .

**Comment by Slakah** [comment]
Not sure why the build is failing:
```
sbt ++2.12.11 test compatibilityCheck
sbt ++2.13.2 test compatibilityCheck
```

Works when running locally.

**Comment by alexarchambault** [comment]
Thanks @Slakah, I'll have a look later today or this week. The `compatibilityCheck` issue shouldn't be a problem (the check should be disabled for the new module, as there's nothing to check compatibility against yet)

**Comment by Slakah** [comment]
@alexarchambault , any updates on this? Don't feel under any pressure though.

**Comment by alexarchambault** [comment]
Sorry this slipped my attention @Slakah, I just pushed a fix.

**Comment by alexarchambault** [comment]
CI is green 🎉, merging!

**Comment by Slakah** [comment]
Awesome, thanks @alexarchambault

**Comment by alexarchambault** [comment]
@Slakah Do you have a Twitter handle, so that I could mention you if I tweet about it?

**Comment by Slakah** [comment]
Yep, my Twitter handle is https://mobile.twitter.com/jrccollier .

**Pull request merged**